src/software/ai/evaluation/calc_best_shot.h

```diff
@@ -20,13 +386,17 @@
  * @param robot_obstacles The locations of any robots on the field that may obstruct
  * the shot. These are treated as circular obstacles, and are usually used to
  * represent robots on the field.
+ * @param goal The goal to shoot at
+ * @param radius The radius for the robot obstacles
  *
  * @return the best target to shoot at and the largest open angle interval for the
  * shot (this is the total angle between the obstacles on either side of the shot
  * vector). If no shot is possible, returns `std::nullopt`
  */
 std::optional<Shot> calcBestShotOnGoal(const Segment &goal_post, const Point &shot_origin,
-                                       const std::vector<Robot> &robot_obstacles);
+                                       const std::vector<Robot> &robot_obstacles,
+                                       TeamType goal,
+                                       double radius = ROBOT_MAX_RADIUS_METERS);
```

Comment:
why did this interface have to change?

Reply:
Depending on what goal we are shooting at the angle mappings differ. I can remove the radius, but I need the goal still

Reply:
OK that's fine then